decks/web-components.mdx

```diff
@@ -0,0 +1,377 @@
+import { Head } from 'mdx-deck'
+
+export { default as theme } from '../theme'
+
+<Head>
+  <title>Web Components</title>
+</Head>
+
+# Web Components
+
+
+by Max Strübing  
+([twitter](https://twitter.com/mxstrbng), [github](https://github.com/mstruebing))
+
+---
+
+Web Components is a suite of different technologies allowing you to create reusable custom elements — with their functionality encapsulated away from the rest of your code — and utilize them in your web apps.
+
+[https://developer.mozilla.org/en-US/docs/Web/Web_Components](https://developer.mozilla.org/en-US/docs/Web/Web_Components)
+
+---
+
+You can create your own custom components i.e: `<my-button>`
+
+---
+
+`-` in the name is mandatory.
+
+---
+
+There are three main concepts.
+
+---
+
+## Custom Element
+
+A set of JS-APIs to define your custom components and their behaviour
+
+---
+
+## Shadow DOM 
+
+API to encapsulate an own DOM-tree to your custom component without 
+interferring with the main-DOM.
+
+---
+
+## HTML templates
+
+`<template>`: is not rendered in the DOM but can be referenced via JS  
+`<slot>`: inject other components into your own component.
+
+---
+
+```js
+customElements.define(name, component);
+```
+
+---
+
+```js
+customElements.define(
+  'my-element',
+  class extends HTMLElement {
+    constructor() {
+      super();
+      this.attachShadow({mode: 'open'});
+
+      this.shadowRoot.innerHTML = `
+              <p>This is my own element</p>
+          `;
+    }
+  },
+);
+```
+
+---
+
+```html
+<my-element />
+```
+
+---
+
+There are many more to extend i.e. `HTMLButtonElement`, `HTMLAudioElement`, `HTMLDivElement` ....
+
+[https://html.spec.whatwg.org/multipage/indices.html#element-interfaces](https://html.spec.whatwg.org/multipage/indices.html#element-interfaces)
+
+---
+
+You can also extend other custom components.
+
+---
+
+## Lifecycle Methods
+
+---
+
+### `connectedCallback` 
+
+- the element has been inserted into the DOM 
+- `componentDidMount`
+
+---
+
+### `disconnectedCallback` 
+
+- the element has been removed from the DOM 
+- right after `componentWillUnmount` 
+
+---
+ 
+### `attributeChangedCallback` 
+
+- everytime a passed property changes 
+- *which is observed
```

Comment:
This looks like a typo to me. Should it be `*which is observed*`?

Reply:
Nope, it was intentionally a star in the presentation. (MDX parses it like I wanted, not how Github does :))